components/cycles/FullCycleOrder.jsx

```diff
@@ -2,16 +2,28 @@ import * as React from 'react';
 import { FlatList } from 'react-native';
 import PropTypes from 'prop-types';
 import ExpandableCycleCard from './CycleOrder';
+import DraggableFlatList from 'react-native-draggable-flatlist';
 
 const FullCycleOrder = (props) => {
-  const { workouts } = props;
+
+  const { 
+    workouts, passWorkoutList 
+  } = props;
+
+  const [workoutList, setWorkoutList] = React.useState(workouts)
+
+  React.useEffect(() => {
+    passWorkoutList(workoutList)
+  }, [workoutList]);
```

Comment:
Can you explain what you're doing with passWorkoutList?

Reply:
Ohhhh nvm I get it

Reply:
Can we rename passWorkoutList to like "SetWorkoutList" or updateWorkoutList

Reply:
yeah useEffect is a hook that basically triggers when state gets updated as a side effect to changing the state. So when the state in this file for the input changes, which is the [workoutList] part, it triggers that which in turn passes the new workoutList state to the CreateCycle file. Bad explanation but that's about all I understand about it

Reply:
setworkoutlist is what I use to set the workoutlist state variable in the hook so that might make it confusing

Reply:
Yeah I think the proper way to do this is with onDragEnd but this looks like it works for now, I'm fine w this going in as is

Reply:
That's what I originally tried, but it threw a warning about mounting a component or something and when I looked into it someone online recommended using the useEffect hook since that is what it's for